.github/workflows/target_workflow.yml

```diff
@@ -0,0 +1,91 @@
+name: QC Target Workflow
+
+on:
+  repository_dispatch:
+    types: [trigger-workflow]
+
+permissions:
+  id-token: write
+  contents: read
+  packages: write
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    environment: main
+    defaults:
+        run:
+            shell: bash
+            working-directory: ./terraform/aws/implementation
+    outputs:
+            tf_env: ${{ steps.set-environment.outputs.tf_env }}
+    steps:
+      - name: Checkout code 
+        uses: actions/checkout@v4
+        with:
+         ref: ${{ github.event.client_payload.sha }}
+
+    
+      - name : Extract Service Name & Version
+        run: |
+          # Parse the service name from the payload
+          service_name=${{ github.event.client_payload.service }}
+          echo "Service Name: $service_name"
```

Comment:
If we are parsing out service name here then this is really a general deployment workflow that can deploy any service not just the Query Connector. However the current workflow name, `QC Target Workflow`, suggestions things are scoped more narrowly to just the Query Connector. Am I following this correctly? Would it make sense to update the name?